instructure/canvas-ios
# Refactor share extension

* Removes presenter
* Exhausts pagination on assignments list so that it doesn't jump around
* Simplifies picking the default course and assignment
* Removes race conditions to hopefully prevent submitting to the wrong
assignment

refs: MBL-14394
affects: student
release note: Improved stability of share extension

Test plan:
- Log in as `s` on narmstrong
- Use a 3rd party app (like Photos) to upload to CS 1400 > File Upload
- Assignments list should not re-order rows as you scroll
